WorkVersionDetailComponent

awead opened the pull request:
Replaces in the in-view logic with a separate component to determine messages displayed to the user. If the user can edit the work, and a draft exists, then the message will give them a link to it.

If the user does not have edit rights, the message only appears if another published version exists.

Fixes #1058

cam156 commented on app/controllers/persons_controller.rb:
```diff
@@ -4,8 +4,26 @@ class PersonsController < ApplicationController
+        qt: 'select',
+        rows: 10
+      )
+      aliases.each do |aliaz|
```
@awead how about agent_alias?  We are never going to know to search for aliaz if we are looking at the code.

cam156 commented on app/controllers/persons_controller.rb:
```diff
@@ -4,8 +4,26 @@ class PersonsController < ApplicationController
+      )
+      aliases.each do |aliaz|
+        person = ActiveFedora::SolrService.query(
+          "id:#{aliaz['name_ssim'].first}",
```
@awead how is the name the ID?  That seems strange

awead replied:
@cam156 This one got me too. It's generated by ActiveFedora's solr name stuff. Specifically, when you define the relationship using the `name` predicate here:

https://github.com/psu-stewardship/scholarsphere/blob/user-object-961/app/models/alias.rb#L6

I would have expected `person_ssim`, but I guess not.

cam156 commented on app/forms/creator_form.rb:
```diff
@@ -30,6 +30,21 @@ def given_name
   end
 
+  def psu_id
+    return unless read_only_name?
```
I'm wondering based on the fact that we are using read_only_name? for things that are not related to name if we should change that to just read_only?

cam156 commented on spec/factories/persons.rb:
```diff
@@ -5,4 +5,12 @@
+    sur_name   'Lately'
+    email      'newkid@example.com'
+    psu_id     'jcl81'
+    orcid_id   '00123445'
```
I wonder if we care about validating the orcid.

awead replied:
@cam156 dunno. I imagine we should. We put that in another ticket and validate by some concrete means, like length, etc., and then look at a future option that could query the orchid service directly, like an autocomplete or just a call back that's done at some point.

cam156 commented on .rubocop_todo.yml:
```diff
@@ -1,6 +1,12 @@
-require: rubocop-rspec
```
@awead did you intend to remove rubocop-rspec?

awead replied:
hmm. no. I think auto-gen did that.

awead replied:
@cam156 `--auto-gen-config` seems to have done that. I'm not sure why it's showing up now, but the require is in `.rubocop_yml` so it doesn't need to be in the todo file.

awead commented:
@cam156 I took a look at the coveralls changes, and the missing coverage doesn't seem to be related to any of the changes in this PR.

awead commented:
@cam156 oohhh-kay.... now it says coverage increased. I rebased it, so maybe that was it.

awead committed:
WorkVersionDetailComponent

Replaces in the in-view logic with a separate component to determine
messages displayed to the user. If the user can edit the work, and a
draft exists, then the message will give them a link to it.

If the user does not have edit rights, the message only appears if
another published version exists.

awead commented:
There's some conflict with #1059 that may render it moot.

## Examples

Published work with available draft

<img width="1254" alt="Screen Shot 2021-06-01 at 3 31 20 PM" src="https://user-images.githubusercontent.com/312085/120380797-9f44f180-c2ef-11eb-91b3-b6a11b61fa3f.png">

Withdrawn work with available  draft

<img width="1254" alt="Screen Shot 2021-06-01 at 3 31 44 PM" src="https://user-images.githubusercontent.com/312085/120380847-b1bf2b00-c2ef-11eb-81ac-44f5525c7d05.png">

Public work with a more recent version

<img width="1254" alt="Screen Shot 2021-06-01 at 3 42 52 PM" src="https://user-images.githubusercontent.com/312085/120381193-1e3a2a00-c2f0-11eb-8c54-85a1941b23e7.png">

Withdrawn version with a current published one

<img width="1254" alt="Screen Shot 2021-06-01 at 3 43 02 PM" src="https://user-images.githubusercontent.com/312085/120381264-3611ae00-c2f0-11eb-96a6-ff962b545047.png">

nicktiberi commented on app/views/resources/_work_version.html.erb:
```diff
@@ -47,14 +47,7 @@
-                  class: 'alert-link' %>
-    </div>
-  <% end %>
+  <%= render WorkVersionDetailComponent.new(work_version: work_version) %>
```
Nice cleanup! Make everything into components! 🙂

nicktiberi commented on app/components/work_version_detail_component.rb:
```diff
@@ -0,0 +1,51 @@
+    return false if current_draft_version?
+
+    i18n_key.present?
+  end
```
Is this a Ruby way of writing this? Just curious. I think more typically I would see something like:

```
def render?
  !current_draft_version? && i18n_key.present?
end
```

Obviously what you have works; I am mainly just wondering if the reason you wrote it that way is due to some convention. Thanks!

awead replied:
@nicktiberi Yeah, that's probably better. This went through a lot of iterations and I wrote it that way so it would look really obvious to me.

nicktiberi reviewed (dismissed):
LGTM! Just have one quick question.

srerickson commented:
@awead - this makes sense to me! Thanks!
